code/game/objects/structures/monster_spawner.dm

```diff
@@ -74,3 +75,86 @@
 	name = "basilisk den"
 	desc = "A den housing a nest of basilisks, bring a coat."
 	mob_types = list(/mob/living/simple_animal/hostile/asteroid/basilisk)
+
+/obj/structure/spawner/sentient
+	var/role_name = "A sentient mob"
+	var/assumed_control_message = "You are a sentient mob from a badly coded spawner"
+
+/obj/structure/spawner/sentient/Initialize(mapload)
+	. = ..()
+	notify_ghosts("\A [name] has been created in \the [get_area(src)]!", source = src, title = "Sentient Spawner Created", flashwindow = FALSE
+	)
+
+/obj/structure/spawner/sentient/on_mob_spawn(mob/created_mob)
+	created_mob.AddComponent(\
+		/datum/component/ghost_direct_control,\
+		role_name = src.role_name,\
+		assumed_control_message = src.assumed_control_message,\
+		after_assumed_control = CALLBACK(src, PROC_REF(became_player_controlled)),\
+	)
+
+/obj/structure/spawner/sentient/proc/became_player_controlled(mob/proteon)
+	return
+
+/obj/structure/spawner/sentient/proteon_spawner
+	name = "eldritch gateway"
+	desc = "A dizzying structure that somehow links into Nar'Sie's own domain. The screams of the damned echo continously."
+	icon = 'icons/obj/cult.dmi'
+	icon_state = "hole"
+	light_power = 2
+	light_color = LIGHT_COLOR_RED
+	max_integrity = 50
+	density = FALSE
+	max_mobs = 2
+	spawn_time = 15 SECONDS
+	mob_types = list(/mob/living/simple_animal/hostile/construct/proteon/hostile)
+	spawn_text = "arises from"
+	faction = list("cult")
+	role_name = "A proteon cult construct"
+	assumed_control_message = null
+	var/gps
```

Comment:
?

Also autodoc, especially one that explains why they have a GPS
```suggestion
	var/obj/item/gps/internal/proteon/gps
```

Reply:
yeah one sec